spot_rl_experiments/spot_rl/envs/place_env.py

```diff
@@ -2,69 +2,216 @@
+                - time_taken (float): Time taken to place the object
+                - success (bool): Whether the place was successful
+                - place_target (np.array([x,y,z])): Place target in base frame
+                - ee_pos (np.array([x,y,z])): End effector position in base frame
+        """
+        success_list = []
+        for place_target in place_target_list:
+            start_time = time.time()
+            if self.use_policies:
+                observations = self.place_env.reset(place_target, is_local)
+                done = False
+
+                while not done:
+                    action = self.policy.act(observations)
+                    observations, _, done, _ = self.place_env.step(arm_action=action)
+
+            else:
+                # Get reset arm position (Is there a better way to do this without using place_env?????????????)
+                self.place_env.reset(place_target, is_local)
+
+                # End effector positions in base frame (as needed by the API)
+                curr_ee_pos = self.place_env.get_gripper_position_in_base_frame()
+                goal_ee_pos = self.place_env.get_corrected_place_target_in_base_frame()
+                intr_ee_pos = generate_intermediate_point(curr_ee_pos, goal_ee_pos)
+
+                # Get direction vector from current ee position to goal ee position for EE orientation
+                dir_rpy_to_goal = get_RPY_from_vector(goal_ee_pos - curr_ee_pos)
+
+                # Go to intermediate point
+                cmd_id = self.spot.move_gripper_to_point(
+                    intr_ee_pos,
+                    dir_rpy_to_goal,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+                # Direct the gripper to face downwards
+                dir_rpy_to_goal = [0.0, np.pi / 2, 0.0]
+
+                # Go to goal point
+                cmd_id = self.spot.move_gripper_to_point(
+                    goal_ee_pos,
+                    dir_rpy_to_goal,
+                    self.config.ARM_TRAJECTORY_TIME_IN_SECONDS,
+                )
+                self.spot.block_until_arm_arrives(cmd_id, timeout_sec=10)
+
+            success_list.append(
+                {
+                    "time_taken": time.time() - start_time,
+                    "success": self.place_env.place_attempted or not self.use_policies,
+                    "place_target": self.place_env.get_corrected_place_target_in_base_frame(),
+                    "ee_pos": self.place_env.get_gripper_position_in_base_frame(),
+                }
+            )
+
+            # Open gripper to drop the object
+            self.spot.open_gripper()
+            # Add sleep as open_gripper() is a non-blocking call
+            time.sleep(1)
+
+        return success_list
+
+    def shutdown(self, should_dock=False):
```

Comment:
make this a common function between all controllers

Reply:
Yes, that will be a separate PR - https://github.com/facebookresearch/spot-sim2real/pull/66